pysgrid/sgrid.py

```diff
@@ -252,6 +267,287 @@ def _save_common_components(self, nc_file):
+        node_x = indices % (self.node_lat.shape[1] - 1)
+        node_y = indices // (self.node_lat.shape[1] - 1)
+        node_ind = np.column_stack((node_y, node_x))
+        if just_one:
+            return node_ind[0]
+        else:
+            return node_ind
+
+    def get_variable_by_index(self, var, index):
+        '''
+        Function to get the node values of a given face index.
+        Emulates the self.grid.nodes[self.grid.nodes.faces[index]] paradigm of unstructured grids
+        '''
+
+        arr = var[:]
+        x = index[:, 0]
+        y = index[:, 1]
+        return np.ma.column_stack((arr[x, y], arr[x + 1, y], arr[x + 1, y + 1], arr[x, y + 1]))
+
+    def build_celltree(self):
+        """
+        Tries to build the celltree for the current UGrid. Will fail if nodes
+        or faces is not defined.
+        """
+        from cell_tree2d import CellTree
+        if self.node_lon is None or self.node_lat is None:
+            raise ValueError(
+                "Nodes must be defined in order to create and use CellTree")
+        if not hasattr(self, '_lin_faces') or not hasattr(self, '_lin_nodes') or self._lin_nodes is None or self._lin_faces is None:
+            self._lin_nodes = np.ascontiguousarray(
+                np.stack((self.node_lon, self.node_lat), axis=-1).reshape(-1, 2))
+            y_size = self.node_lon.shape[0]
+            x_size = self.node_lon.shape[1]
+            self._lin_faces = np.array([np.array([[x, x + 1, x + x_size + 1, x + x_size]
+                                                  for x in range(0, x_size - 1, 1)]) + y * x_size for y in range(0, y_size - 1)])
+            self._lin_faces = np.ascontiguousarray(
+                self._lin_faces.reshape(-1, 4).astype(np.int32))
+        self._tree = CellTree(self._lin_nodes, self._lin_faces)
+
+    def interpolate_var_to_points(self, points, variable, indices=None, alphas=None, mask=None):
+
+        ind = indices
+        if ind is None:
+            ind = self.locate_faces(points)
+        translation = self.infer_grid(variable)
+        lons = self.node_lon[:]
+        lats = self.node_lat[:]
+        if translation is not None:
+            if translation == 'face':
+                lons, lats = self.center_lon[:], self.center_lat[:]
+            if translation == 'edge1':
+                lons, lats = self.edge1_lon[:], self.edge1_lat[:]
+            if translation == 'edge2':
+                lons, lats = self.edge2_lon[:], self.edge2_lat[:]
+            ind = self.translate_index(points, ind, lons, lats, translation)
+
+        if alphas is None:
+            alphas = self.interpolation_alphas(points, ind, lons, lats)
+        vals = self.get_variable_by_index(variable, ind)
+        result = np.ma.sum(vals * alphas, axis=1)
+        if mask is not None:
+            # REVISIT LATER
+            result.mask = mask[ind[:, 0], ind[:, 1]]
+        return result
+
+    def infer_grid(self, variable):
```

Comment:
I am unsure how this will be used.  Maybe a few examples would help.  Maybe, if the goal is to find elements at nodes, edges, and faces, we should always ask for that information in a keyword  rather than inferring.